### PR TITLE
Add a configuration to set ufs fetch timeout

### PR DESCRIPTION
**Pull request by jiacheliu3**
### What changes are proposed in this pull request?

This change makes the UFS fetch operation timeout configurable.

Also a minor change which uses the sized constructor for HashMap.

### Why are the changes needed?

Some UFSes can be bad in worst response time so the fetch operation can timeout easily when the timeout is just 100ms. This might be most useful in the development stage, where the user is accessing a remote UFS like S3 from a random location, so the response time can be randomly bad.

### Does this PR introduce any user facing changes?

This adds a new property key for the user to configure if really needs to.

### CODE REVIEW

File: core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 1]
```diff
@@ -2811,6 +2811,14 @@ public String toString() {
           .setScope(Scope.MASTER)
           .setConsistencyCheckLevel(ConsistencyCheckLevel.WARN)
           .build();
+  public static final PropertyKey MASTER_METADATA_SYNC_UFS_PREFETCH_TIMEOUT =
+      new Builder(Name.MASTER_METADATA_SYNC_UFS_PREFETCH_TIMEOUT)
+          .setDefaultValue("100ms")
+          .setDescription("The timeout for a metadata fetch operation from the UFSes. "
+              + "Adjust this timeout according to the expected UFS worse-case response time.")
```
Comment (ZhuTopher) [inline core/common/src/main/java/alluxio/conf/PropertyKey.java:2818]: `worst-case` instead of `worse-case`?

**Review by ZhuTopher — APPROVED** [review]
Small typo maybe? Otherwise LGTM

**Comment by jiacheliu3** [comment]
The failed tests are flakes irrelevant to this change
```
Error: 8.480 [ERROR] alluxio.server.tieredstore.TierPromoteIntegrationTest.promoteByTierSwap[0]  Time elapsed: 5.208 s  <<< ERROR!
java.util.concurrent.TimeoutException: Timed out waiting for file is not in memory options: WaitForOptions{interval=10, timeout=2000} last value: false
	at alluxio.util.CommonUtils.waitForResult(CommonUtils.java:384)
	at alluxio.util.CommonUtils.waitFor(CommonUtils.java:342)
	at alluxio.server.tieredstore.TierPromoteIntegrationTest.promoteByTierSwap(TierPromoteIntegrationTest.java:116)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at alluxio.testutils.LocalAlluxioClusterResource$1.evaluate(LocalAlluxioClusterResource.java:202)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:288)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:282)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Error: 2.739 [ERROR] alluxio.client.cli.job.DistributedCommandsStatsTest.testAsyncPersistCancelStats  Time elapsed: 45.341 s  <<< ERROR!
java.util.concurrent.TimeoutException: Timed out waiting for job 1646809886202 to be one of status [CANCELED] options: WaitForOptions{interval=20, timeout=45000} last value: false
	at alluxio.util.CommonUtils.waitForResult(CommonUtils.java:384)
	at alluxio.util.CommonUtils.waitFor(CommonUtils.java:342)
	at alluxio.job.util.JobTestUtils.waitForJobStatus(JobTestUtils.java:96)
	at alluxio.client.cli.job.DistributedCommandsStatsTest.testAsyncPersistCancelStats(DistributedCommandsStatsTest.java:197)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:258)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at alluxio.testutils.LocalAlluxioClusterResource$ResetRule$1.evaluate(LocalAlluxioClusterResource.java:319)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:288)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:282)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**